### DISCUSSION TITLE
使用过程中有任何疑问,都可以到这里来交流!

### DISCUSSION THREAD

**Discussion by xuexiangjys** [xuexiangjys/XUpdate#125] [Q&A]
## 👋 Welcome!
这里是xupdate官方的交流社区,在使用过程中遇到任何疑问,欢迎到这里来交流讨论!

## 遇到问题如何解决?

这里我简单提供几个解决问题的方法:

* 查阅文档,寻找相关的使用说明:https://github.com/xuexiangjys/XUpdate/wiki
* 查阅常见问题,是否有类似问题的解决方法: https://github.com/xuexiangjys/XUpdate/wiki/常见问题
* 观看XUpdate系列视频: https://space.bilibili.com/483850585/channel/detail?cid=164280
* 在XUpdate的官方Discussions中寻求帮助: https://github.com/xuexiangjys/XUpdate/discussions
* 如果确认是问题,可在[issue里面按模板填写出错信息](https://github.com/xuexiangjys/XUpdate/issues/new?assignees=&labels=&template=--bug--.md&title=),我将在一周之内予以解决!
* 加官方交流QQ群: 720212425

**Comment by mynameispp** [comment]
后台返回的数据中ApkMd5的值是用DefaultFileEncryptor.encryptFile(File file)获取的吗？

**Reply by xuexiangjys** [reply]
是的

**Comment by Yonion** [comment]
提示“更新失败：已经被忽略的版本！”，可以在别的地方取消这个忽略设置吗？需求是在别的地方要重新检测下载

**Reply by xuexiangjys** [reply]
为何会有如此前后矛盾的需求？建议关闭忽略版本的功能

**Reply by Yonion** [reply]
在主页弹窗中点击 忽略该版本 后进入主页不再弹该版本，但是在设置页面检测新版本的时候要弹出来该版本进行升级，这边自定义的解析，一样的请求数据，发现一开始如果点击了忽略版本 那么后续再设置setIsIgnorable(false)也不会再弹出来了

**Reply by xuexiangjys** [reply]
这里确实可以优化一下

**Reply by xuexiangjys** [reply]
在最新的2.1.2版本上可以实现你的需求

**Reply by Yonion** [reply]
可以了，感谢大佬！

**Comment by lookapu** [comment]
`if (updateInfo.versionCode! <= int.parse(packageInfo.buildNumber)) {
        hasUpdate = false;
      }`

@xuexiangjys 
请问这里能否再增加一个版本号的判断，如果只判断这个，比如1.0.3+10，无法升级到2.0.1+5 ~~~

谢谢啦~~~

**Reply by xuexiangjys** [reply]
这里只是本地校验, 还是应当以云端校验为主
